### PR TITLE
use unique_ptr and test both pointer modes

### PR DESCRIPTION
**Pull request by amcamd**
refactor tests
* use std::unique_ptr for hipMalloc and hipFree
* use std::unique_ptr for rocblas_create_handle and rocblas_destroy_handle
* all tests for rocblas_pointer_mode_host and rocblas_pointer_mode_device

### CODE REVIEW

File: clients/gtest/blas1_gtest.cpp [hunk 1]
```diff
@@ -211,18 +223,7 @@ TEST_P(blas1_gtest, axpy_float)
...
-            EXPECT_EQ(rocblas_status_invalid_size, status);
-        }
-        else if( arg.incx < 0){
-            EXPECT_EQ(rocblas_status_invalid_size, status);
-        }
-        else if( arg.incy < 0){
-            EXPECT_EQ(rocblas_status_invalid_size, status);
-        }
-    }
+    EXPECT_EQ(rocblas_status_success, status);
```
Comment (kknox) [inline clients/gtest/blas1_gtest.cpp]: Does this reduce test precision?  Looks like the old code tried to decipher that the returned error code was a particular value.

File: clients/gtest/blas1_gtest.cpp [hunk 2]
```diff
@@ -174,13 +175,24 @@ TEST_P(blas1_gtest, iamax_float)
...
+    // GetParam return a tuple. Tee setup routine unpack the tuple
+    // and initializes arg(Arguments) which will be passed to testing routine
+    // The Arguments data struture have physical meaning associated.
+    // while the tuple is non-intuitive.
+    Arguments arg = setup_blas1_arguments( GetParam() );
+    rocblas_status status = testing_iamax<double>( arg );
+    // if not success, then the input argument is problematic, so detect the error message
+    if(status != rocblas_status_success)
+    {
+        EXPECT_EQ(rocblas_status_success, status);
```
Comment (kknox) [inline clients/gtest/blas1_gtest.cpp]: Do we need to protect the EXPECT_EQ with a branch?

File: clients/gtest/blas1_gtest.cpp (hunk 1 above)
Comment (kknox) [reply clients/gtest/blas1_gtest.cpp]: Maybe you meant to use your new macro CHECK_ROCBLAS_ERROR()?

File: clients/gtest/blas1_gtest.cpp (hunk 2 above)
Comment (kknox) [reply clients/gtest/blas1_gtest.cpp]: Possibly, you meant to use your new macro CHECK_ROCBLAS_ERROR()

Comment (amcamd) [reply clients/gtest/blas1_gtest.cpp]: Thanks, you are correct, will remove branch and use CHECK_ROCBLAS_ERROR()

**Pull request merged**